btsieve/src/bitcoin/bitcoin_d_http_blocksource.rs

```diff
@@ -0,0 +1,205 @@
+    BlockDeserialization(String),
+    TransactionDeserialization(String),
+}
+
+#[derive(Clone)]
+pub struct BitcoindHttpBlockSource {
+    network: Network,
+    base_url: String,
+    client: Client,
+}
+
+impl BitcoindHttpBlockSource {
+    pub fn new(url: String, network: Network, client: Client) -> Self {
+        Self {
+            network,
+            base_url: url,
+            client,
+        }
+    }
+
+    fn latest_block(&self) -> impl Future<Item = Block, Error = Error> + Send + 'static {
+        let cloned_self = self.clone();
+
+        self.latest_block_hash()
+            .and_then(move |latest_block_hash| cloned_self.block_by_hash(latest_block_hash))
+    }
+
+    fn latest_block_hash(&self) -> impl Future<Item = String, Error = Error> + Send + 'static {
+        let bitcoind_blockchain_info_url = format!("{}/rest/chaininfo.json", self.base_url);
+
+        self.client
+            .get(bitcoind_blockchain_info_url.as_str())
+            .send()
+            .inspect(|something| {
+                log::debug!("something: {:?}", something);
+            })
+            .map_err(|e| {
+                log::debug!("inside error!");
+                Error::Reqwest(e)
+            })
+            .inspect(|something| {
+                log::debug!("something else: {:?}", something);
+            })
+            .and_then(move |mut response| {
+                response
+                    .json::<BitcoindBlockchainInfo>()
+                    .map_err(Error::Reqwest)
+            })
+            .map(move |blockchain_info| blockchain_info.bestblockhash)
+    }
+
+    fn block_by_hash(
+        &self,
+        block_hash: String,
+    ) -> impl Future<Item = Block, Error = Error> + Send + 'static {
+        let raw_block_by_hash_url = format!("{}/rest/block/{}.hex", self.base_url, block_hash);
+
+        self.client
+            .get(raw_block_by_hash_url.as_str())
+            .send()
+            .map_err(Error::Reqwest)
+            .and_then(|mut response| response.text().map_err(Error::Reqwest))
+            .and_then(move |response_text| {
+                let mut response_text: String = response_text;
+
+                // for whatever reason there can be line-breaks in the hex responses (tested on
+                // regtest), thus we remove them
+                if response_text.ends_with('\n') {
+                    response_text.pop();
+                    if response_text.ends_with('\r') {
+                        response_text.pop();
+                    }
+                }
+                hex::decode(response_text).map_err(Error::Hex)
+            })
+            .and_then(|bytes| {
+                deserialize(bytes.as_ref()).map_err(|e| {
+                    log::error!("Got new block but failed to deserialize it because {:?}", e);
+                    Error::BlockDeserialization(format!(
+                        "Failed to deserialize the response from bitcoin-d into a block: {}",
+                        e
```

Comment:
I suggest to just store the error inside `Error::BlockDeserialization` as you do for the other variants as well :)